musmart/melody/contour/step_contour.py

```diff
@@ -1,260 +1,161 @@
+            Length of the output step contour vector (default is 64)
+
+        References
+        ----------
+        [1] Müllensiefen, D. (2009). Fantastic: Feature ANalysis Technology Accessing
+        STatistics (In a Corpus): Technical Report v1.5
+        """
+        if len(pitches) != len(durations):
+            raise ValueError(
+                f"The length of pitches (currently {len(pitches)}) must be equal to "
+                f"the length of durations (currently {len(durations)})"
+            )
+
+        self._step_contour_length = step_contour_length
+        self._contour = self._calculate_contour(pitches, durations)
+
+    def _normalize_durations(self, durations: list[float]) -> list[float]:
+        """Helper function to normalize note durations to fit within 4 bars of 4/4 time
+        (64 tatums total).
+
+        Parameters
+        ----------
+        durations : list[float]
+            List of duration values measured in tatums
+
+        Returns
+        -------
+        list[float]
+            List of normalized duration values
+        """
+        total_duration = sum(durations)
+        if total_duration == 0:
+            return durations
+
+        normalized = [
+            self._step_contour_length * (duration / total_duration)
+            for duration in durations
+        ]
+        return normalized
+
+    def _expand_to_vector(
+        self,
+        pitches: list[int],
+        normalized_durations: list[float]
+    ) -> list[int]:
+        """Helper function to create a vector of length step_contour_length by repeating
+        each pitch value proportionally to its normalized duration.
+
+        Parameters
+        ----------
+        pitches : list[int]
+            List of pitch values
+        normalized_durations : list[float]
+            List of normalized duration values (should sum to step_contour_length)
+
+        Returns
+        -------
+        list[int]
+            List of length step_contour_length containing repeated pitch values
+        """
+        result = []
+        for pitch, duration in zip(pitches, normalized_durations):
+            repetitions = round(duration)
+            result.extend([pitch] * repetitions)
+
+        if len(result) > self._step_contour_length:
+            result = result[:self._step_contour_length]
+        elif len(result) < self._step_contour_length:
+            result.extend([result[-1]] * (self._step_contour_length - len(result)))
+
+        return result
+
+    def _calculate_contour(
+        self,
+        pitches: list[int],
+        durations: list[float]
+    ) -> list[int]:
+        """Calculate the step contour from input pitches and durations."""
+        normalized_durations = self._normalize_durations(durations)
+        return self._expand_to_vector(pitches, normalized_durations)
+
+    @property
+    def contour(self) -> list[int]:
+        """Get the step contour vector."""
+        return self._contour
```

Comment:
Why do we need both `_contour` and `contour?

Reply:
This was suggested by cursor - the motivation is that `_contour` is immutable so won't be overwritten by the user after initialisation (either in error or on purpose). Then `contour` gives a public interface for the encapsulated `_contour`

I don't have strong feelings about it, but it feels like good practice!

Reply:
I see! No, I'd prefer to remove this, if we did that everywhere it would make things very verbose.

Reply:
Understood! fixed